url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/StepRun.java

```diff
@@ -0,0 +1,393 @@
+    public static boolean callbbeq = false;
+    public static boolean stepruncomment = false;
+    public static int end_pos;
+    public static int transfer;
+    public static int srun = 0;
+    public static boolean commentflag = false;
+
+    public StepRun(File path, int ini, int fin, int f)
+      {
+
+
+
+        t = new Thread(this, "steprun");
+        fpath = path;
+
+        if (f == 0 && ini != 0)
+        {
+            ini_line = ini;
+            fin_line = fin;
+            if (ini == fin)
+            {
+                ini_line = 1;
+            } else
+            {
+                ini_line = ini;
+            }
+            //System.out.println("yepii " + ini_line + "BLAH" + fin_line + "BLAH" + f);
+        }
+
+        f = 1;
+        //System.out.println(" \n" + ini + "" + fin + "" + ini_line + " " + fin_line + " in steprun");
+
+      }
+
+    public void scanStart()
+      {
+        // linePainter = new LinePainter(frontend.FrontEnd.activepane);
+
+        FrontEnd.exceptionraised = 0;
+        try{
+       // System.err.println("step run scanstart()");
+        if (ini_line != 0)
+        {
+           // System.out.println("STEP RUN UPAR");
+           
+
+                ScanFile.pos = ScanFile.br.getFilePointer();
+                initial_pos = ScanFile.pos.intValue();
+
+                //System.out.print("val returned is :" + initial_pos + " and length is :" + ScanFile.path.length());
+                if (ini_line < fin_line)
+                {
+                    ScanFile.br.seek(ScanFile.pos);
+                } else
+                {
+                    //System.out.println(" \n stopping step run ");
+                    callbbeq = false;
+                    commentflag = false;
+                    frontend.FrontEnd.steprun.setEnabled(false);
+                    frontend.FrontEnd.warning.setVisible(true);
+                    frontend.FrontEnd.cross.setVisible(true);
+                    frontend.FrontEnd.cross.setEnabled(true);
+                    frontend.FrontEnd.run_stepOut.setEnabled(false);
+                    frontend.FrontEnd.run_stepOver.setEnabled(false);
+                    frontend.FrontEnd.r_stepOut.setEnabled(false);
+                    frontend.FrontEnd.r_stepover.setEnabled(false);
+                    frontend.FrontEnd.check.removeTabAt(0);
+
+                    FrontEnd.reset.setEnabled(false);
+                    return;
+                }
+                String strLine = "";
+                if (!callbbeq)
+                {
+                    if (initial_pos == 0 & ScanFile.mainPresent)
+                    {
+                        backend.ScanFile.br.seek(ScanFile.mainPosition);
+                        backend.ScanFile.pos = ScanFile.mainPosition;
+                        initial_pos = ScanFile.pos.intValue();
+                    } else
+                    {
+                       // initial_pos++;
+                    }
+                }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 91:102, 227:241.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.